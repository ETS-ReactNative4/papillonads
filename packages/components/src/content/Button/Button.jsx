import React from 'react'
import cx from 'classnames'
import { CSSTransition } from 'react-transition-group'
import { propTypes, defaultProps } from './Button.prop'
import styles from './Button.scss'
import { Icon } from '../Icon'
import { TabFocusTrap } from '../TabFocusTrap'
import { Spinner } from '../Spinner'

export function Button({
  className,
  children,
  icon,
  buttonType,
  size,
  isFullWidth,
  onBlur,
  onClick,
  loading,
  disabled,
  indicateDropdown,
  href,
  type,
  isActive,
  ...otherProps
}) {
  const classNames = cx(styles.Button, className, styles[`Button--${buttonType}`], {
    [styles['Button--disabled']]: disabled,
    [styles[`Button--${size}`]]: size,
    [styles['Button--full-width']]: isFullWidth,
    [styles['Button--is-active']]: isActive,
  })

  const iconColor = buttonType === 'muted' || buttonType === 'naked' ? 'secondary' : 'white'

  const Element = href ? 'a' : 'button'

  return (
    <Element
      onBlur={(e) => {
        if (onBlur && !disabled) {
          onBlur(e)
        }
      }}
      onClick={(e) => {
        if (onClick && !disabled && !loading) {
          onClick(e)
        }
      }}
      className={classNames}
      disabled={disabled}
      href={!disabled ? href : null}
      type={type}
      {...otherProps}
    >
      <TabFocusTrap className={styles['Button__inner-wrapper']}>
        {icon && <Icon className={styles.Button__icon} size={size === 'small' ? 'tiny' : 'small'} icon={icon} color={iconColor} />}
        {children && <span className={styles.Button__label}>{children}</span>}
        {indicateDropdown && <Icon className={styles['Button__dropdown-icon']} icon="ArrowDown" color={iconColor} />}
        <CSSTransition
          in={loading}
          timeout={1000}
          classNames={{
            enter: styles['Button--spinner--enter'],
            enterActive: styles['Button--spinner-active'],
            exit: styles['Button--spinner--exit'],
            exitActive: styles['Button--spinner-exit-active'],
          }}
          mountOnEnter
          unmountOnExit
        >
          <Spinner
            className={styles.Button__spinner}
            size="small"
            color={buttonType === 'muted' || buttonType === 'naked' ? 'default' : 'white'}
          />
        </CSSTransition>
      </TabFocusTrap>
    </Element>
  )
}

Button.propTypes = propTypes

Button.defaultProps = defaultProps
