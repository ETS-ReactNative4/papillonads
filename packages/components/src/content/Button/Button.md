# Button

Buttons communicate the action that will occur when the user clicks them. They make actions immediately visible.

## When to use

When you want to communicate an action to a user. There are a number of types to change the appearance of a button. Here is a list of when to use them:

- **Primary** - Used for the most important actions in any scenario. Donâ€™t use more than one primary button in a section or screen to avoid overwhelming users.
- **Muted** - Used for a secondary actions, the most commonly used button type.
- **Positive** - For use when the action has a positive connotation such as creating or publishing a new entity.
- **Negative** - For destructive actions - when something can't be undone. For example, deleting entities.

## Best practices

- Use clear, succinct, and accurate labels
- Start button labels with verbs, for example "Add field", not "New field"
- Position buttons in consistent places in the interface
- Reduce complexity by using a small number of actions. Too many actions can create confusion when having to decide
