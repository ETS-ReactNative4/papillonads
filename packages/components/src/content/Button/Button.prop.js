import { FocusEventHandler, MouseEventHandler } from 'react'
import { string, bool, oneOf, node } from 'prop-types'
import { propTypes as iconPropTypes } from '../Icon/Icon.prop'

export const buttonType = {
  primary: 'primary',
  positive: 'positive',
  negative: 'negative',
  muted: 'muted',
  naked: 'naked',
}

export const type = {
  button: 'button',
  submit: 'submit',
  reset: 'reset',
}

export const buttonSize = {
  small: 'small',
  large: 'large',
}

export const propTypes = {
  className: string,
  icon: iconPropTypes.icon,
  indicateDropdown: bool,
  onClick: MouseEventHandler,
  isFullWidth: bool,
  onBlur: FocusEventHandler,
  loading: bool,
  disabled: bool,
  buttonType: oneOf(Object.keys(buttonType)),
  type: oneOf(Object.keys(type)),
  size: oneOf(Object.keys(buttonSize)),
  href: string,
  children: node,
  isActive: bool,
}

/* istanbul ignore next */
export const defaultProps = {
  className: null,
  icon: null,
  indicateDropdown: false,
  onClick: () => {},
  isFullWidth: false,
  onBlur: () => {},
  loading: false,
  disabled: false,
  buttonType: buttonType.primary,
  type: type.button,
  size: buttonSize.small,
  href: null,
  children: null,
  isActive: false,
}
