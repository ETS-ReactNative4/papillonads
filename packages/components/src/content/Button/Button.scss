/* stylelint-disable no-descending-specificity */
@import '~@papillonads/css/build/content/settings/colors';
@import '~@papillonads/css/build/content/settings/typography';
@import '~@papillonads/css/build/content/settings/transitions';
@import '~@papillonads/css/build/content/settings/shadows';
@import '~@papillonads/css/build/content/settings/z-index';

.Button:focus {
  // composes: focus-border--default from a11y.scss
  outline: none;
  border: 1px solid $color-primary;
  box-shadow: $glow-primary;
}

.Button,
.Button:link {
  box-sizing: border-box;
  height: calc(1rem * (40 / #{$font-base-default}));
  display: inline-block;
  padding: 0;
  border: calc(1rem * (1 / #{$font-base-default})) solid #{$color-element-dark};
  border-radius: calc(1rem * (2 / #{$font-base-default}));
  font-family: $font-stack-primary;
  font-size: calc(1rem * (14 / #{$font-base-default}));
  overflow: hidden;
  background-size: 100% 200%;
  transition: background $transition-duration-default $transition-easing-default,
    opacity $transition-duration-default $transition-easing-default, border-color $transition-duration-default $transition-easing-default;
  vertical-align: middle;
  text-decoration: none;

  &:hover:not(.Button--disabled) {
    cursor: pointer;
  }
}

.Button__icon {
  min-width: calc(1rem * (18 / #{$font-base-default}));
}

.Button--full-width {
  width: 100%;
}

.Button--muted {
  background-color: $color-element-light;

  &:hover:not(.Button--disabled),
  &:focus:not(.Button--disabled) {
    background-color: $color-element-mid;
  }

  &:active:not(.Button--disabled),
  &.Button--is-active:not(.Button--disabled) {
    transition: none;
    background-color: $color-element-dark;
  }
}

.Button--naked {
  background-color: transparent;
  border: none;

  &:hover:not(.Button--disabled),
  &:focus:not(.Button--disabled) {
    background-color: $color-element-light;
  }

  &:active:not(.Button--disabled),
  &.Button--is-active:not(.Button--disabled) {
    transition: none;
    background-color: $color-element-mid;
  }

  &.Button:link {
    border: none;
  }
}

.Button--primary {
  border-color: $color-blue-dark;
  background-image: linear-gradient(to top, $color-blue-base, $color-blue-light);
  background-color: $color-blue-base;
  background-size: 100% 200%;

  &:hover:not(.Button--disabled),
  &:focus:not(.Button--disabled) {
    background-position: 0 100%;
  }

  &:focus:not(.Button--disabled) {
    border-color: $color-blue-dark;
    box-shadow: $glow-primary;
  }

  &:active:not(.Button--disabled),
  &.Button--is-active:not(.Button--disabled) {
    transition: none;
    background-image: none;
    background-color: $color-blue-dark;
  }

  & .Button__label {
    color: $color-element-lightest;
  }

  &.Button:link {
    border-color: $color-blue-dark;
  }
}

.Button--positive {
  border-color: $color-green-dark;
  background-image: linear-gradient(to top, $color-green-base, $color-green-light);
  background-size: 100% 200%;

  &:hover:not(.Button--disabled),
  &:focus:not(.Button--disabled) {
    background-position: 0 100%;
  }

  &:focus:not(.Button--disabled) {
    border-color: $color-green-dark;
    box-shadow: $glow-positive;
  }

  &:active:not(.Button--disabled),
  &.Button--is-active:not(.Button--disabled) {
    transition: none;
    background-image: none;
    background-color: $color-green-dark;
  }

  & .Button__label {
    color: $color-element-lightest;
  }

  &.Button:link {
    border-color: $color-green-dark;
  }
}

.Button--negative {
  border-color: $color-red-dark;
  background-image: linear-gradient(to top, $color-red-base, $color-red-light);
  background-size: 100% 200%;

  &:hover:not(.Button--disabled),
  &:focus:not(.Button--disabled) {
    background-position: 0 100%;
  }

  &:focus:not(.Button--disabled) {
    border-color: $color-red-dark;
    box-shadow: $glow-negative;
  }

  &:active:not(.Button--disabled),
  &.Button--is-active:not(.Button--disabled) {
    transition: none;
    background-image: none;
    background-color: $color-red-dark;
  }

  & .Button__label {
    color: $color-element-lightest;
  }

  &.Button:link {
    border-color: $color-red-dark;
  }
}

.Button--warning {
  border-color: $color-orange-dark;
  background-image: linear-gradient(to top, $color-orange-base, $color-orange-light);
  background-size: 100% 200%;

  &:hover:not(.Button--disabled),
  &:focus:not(.Button--disabled) {
    background-position: 0 100%;
  }

  &:focus:not(.Button--disabled) {
    border-color: $color-orange-dark;
    box-shadow: $glow-warning;
  }

  &:active:not(.Button--disabled),
  &.Button--is-active:not(.Button--disabled) {
    transition: none;
    background-image: none;
    background-color: $color-orange-dark;
  }

  & .Button__label {
    color: $color-element-lightest;
  }

  &.Button:link {
    border-color: $color-orange-dark;
  }
}

.Button--disabled {
  opacity: 0.5;
  &:hover {
    cursor: not-allowed;
  }
}

.Button__inner-wrapper {
  display: flex;
  height: 100%;
  z-index: $z-index-default;
  padding: 0 calc(1rem * (14 / #{$font-base-default}));
  align-items: center;
  justify-content: center;
}

.Button__label {
  margin: 0 calc(1rem * (4 / #{$font-base-default}));
  color: $color-text-mid;
  line-height: 2;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.Button--small,
.Button--small:link {
  height: calc(1rem * (30 / #{$font-base-default}));

  & .Button__inner-wrapper {
    padding: 0 calc(1rem * (10 / #{$font-base-default}));
  }

  & .Button__label {
    font-size: calc(1rem * (14 / #{$font-base-default}));
  }
}

.Button--large {
  height: calc(1rem * (48 / #{$font-base-default}));
  & .Button__label {
    line-height: 1;
    margin: 0.25rem;
    font-size: 1.15rem;
  }
}

.Button__spinner {
  transition: opacity, width $transition-duration-default $transition-easing-default;
}

.Button--spinner--enter {
  opacity: 0;
  width: 0;
}

.Button--spinner-active {
  opacity: 1;
  width: 14px;
}

.Button--spinner--exit {
  opacity: 1;
  width: 14px;
}

.Button--spinner-exit-active {
  opacity: 0;
  width: 0;
}
/* stylelint-enable no-descending-specificity */
