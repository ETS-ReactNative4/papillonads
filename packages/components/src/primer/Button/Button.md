# Button

- Default Button
- Default Link Button

> Buttons are used for actions, like in forms, while textual hyperlinks are used for destinations, or moving from one page to another.

## Default Button

Use the standard—yet classy—.btn for form actions and primary page actions. These are used extensively around the site.

When using a `<button>` element, always specify a type. When using a `<a>` element, always add role="button" for accessibility.

## Default Link Button

📝 Coming soon 🐬
