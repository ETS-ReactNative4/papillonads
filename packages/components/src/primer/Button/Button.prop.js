import { string, oneOf, shape, func } from 'prop-types'
import { iconName } from '../Icon/Icon.prop'

export const buttonElement = {
  button: 'button',
  a: 'a',
}

export const buttonVariant = {
  default: 'default',
  primary: 'primary',
  danger: 'danger',
  outline: 'outline',
  blue: 'blue',
  orange: 'orange',
}

export const buttonSize = {
  small: 'small',
  normal: 'normal',
  large: 'large',
}

export const buttonState = {
  enabled: 'enabled',
  disabled: 'disabled',
}

export const iconAlignment = {
  left: 'left',
  right: 'right',
}

export const buttonInputType = {
  file: 'file',
}

export const propTypes = {
  className: string,
  element: oneOf(Object.keys(buttonElement)),
  href: string,
  text: string.isrequired,
  variant: oneOf(Object.keys(buttonVariant)),
  size: oneOf(Object.keys(buttonSize)),
  state: oneOf(Object.keys(buttonState)),
  icon: shape({
    name: oneOf(Object.keys(iconName)).isRequired,
    alignment: oneOf(Object.keys(iconAlignment)),
  }),
  onClick: func.isRequired,
  inputType: oneOf(Object.keys(buttonInputType)),
}

export const defaultProps = {
  className: null,
  element: buttonElement.button,
  href: null,
  variant: buttonVariant.default,
  size: buttonSize.normal,
  state: buttonState.enabled,
  icon: null,
  hasInput: false,
  inputType: null,
}
