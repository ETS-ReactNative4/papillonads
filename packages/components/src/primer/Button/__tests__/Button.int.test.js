import {
  renderCustom,
  renderDefaultButton,
  renderDefaultSmallButton,
  renderDefaultLargeButton,
  renderDefaultDisabledButton,
  renderDefaultIconButtonLeftAligned,
  renderDefaultIconButtonRightAligned,
  renderDefaultLinkButton,
  renderPrimaryButton,
  renderPrimarySmallButton,
  renderPrimaryLargeButton,
  renderPrimaryDisabledButton,
  renderPrimaryIconButtonLeftAligned,
  renderPrimaryIconButtonRightAligned,
  renderDangerButton,
  renderDangerSmallButton,
  renderDangerLargeButton,
  renderDangerDisabledButton,
  renderDangerIconButtonLeftAligned,
  renderDangerIconButtonRightAligned,
  renderOutlineButton,
  renderOutlineSmallButton,
  renderOutlineLargeButton,
  renderOutlineDisabledButton,
  renderOutlineIconButtonLeftAligned,
  renderOutlineIconButtonRightAligned,
  renderBlueButton,
  renderBlueSmallButton,
  renderBlueLargeButton,
  renderBlueDisabledButton,
  renderBlueIconButtonLeftAligned,
  renderBlueIconButtonRightAligned,
  renderOrangeButton,
  renderOrangeSmallButton,
  renderOrangeLargeButton,
  renderOrangeDisabledButton,
  renderOrangeIconButtonLeftAligned,
  renderOrangeIconButtonRightAligned,
} from './Button.int.render'
import { buttonElement } from '../Button.prop'

describe('<Button />', () => {
  const onClickMockFn = jest.fn()

  describe('Event', () => {
    test('must call onClick() when element button', () => {
      const mountCustomRender = global.renderMount(renderCustom(buttonElement.button, onClickMockFn))
      mountCustomRender.find('button').first().simulate('click')
      expect(mountCustomRender.props().onClick).toBe(onClickMockFn)
      expect(mountCustomRender.props().onClick).toHaveBeenCalled()
    })

    test('must call onClick() when element a', () => {
      const mountCustomRender = global.renderMount(renderCustom(buttonElement.a, onClickMockFn))
      mountCustomRender.find('a').first().simulate('click')
      expect(mountCustomRender.props().onClick).toBe(onClickMockFn)
      expect(mountCustomRender.props().onClick).toHaveBeenCalled()
    })
  })

  describe('Render', () => {
    test('must match renderDefaultButton()', () => {
      expect(global.renderToJSON(renderDefaultButton())).toMatchSnapshot()
    })

    test('must match renderDefaultSmallButton()', () => {
      expect(global.renderToJSON(renderDefaultSmallButton())).toMatchSnapshot()
    })

    test('must match renderDefaultLargeButton()', () => {
      expect(global.renderToJSON(renderDefaultLargeButton())).toMatchSnapshot()
    })

    test('must match renderDefaultDisabledButton()', () => {
      expect(global.renderToJSON(renderDefaultDisabledButton())).toMatchSnapshot()
    })

    test('must match renderDefaultIconButtonLeftAligned()', () => {
      expect(global.renderToJSON(renderDefaultIconButtonLeftAligned())).toMatchSnapshot()
    })

    test('must match renderDefaultIconButtonRightAligned()', () => {
      expect(global.renderToJSON(renderDefaultIconButtonRightAligned())).toMatchSnapshot()
    })

    test('must match renderDefaultLinkButton()', () => {
      expect(global.renderToJSON(renderDefaultLinkButton())).toMatchSnapshot()
    })

    test('must match renderPrimaryButton()', () => {
      expect(global.renderToJSON(renderPrimaryButton())).toMatchSnapshot()
    })

    test('must match renderPrimarySmallButton()', () => {
      expect(global.renderToJSON(renderPrimarySmallButton())).toMatchSnapshot()
    })

    test('must match renderPrimaryLargeButton()', () => {
      expect(global.renderToJSON(renderPrimaryLargeButton())).toMatchSnapshot()
    })

    test('must match renderPrimaryDisabledButton()', () => {
      expect(global.renderToJSON(renderPrimaryDisabledButton())).toMatchSnapshot()
    })

    test('must match renderPrimaryIconButtonLeftAligned()', () => {
      expect(global.renderToJSON(renderPrimaryIconButtonLeftAligned())).toMatchSnapshot()
    })

    test('must match renderPrimaryIconButtonRightAligned()', () => {
      expect(global.renderToJSON(renderPrimaryIconButtonRightAligned())).toMatchSnapshot()
    })

    test('must match renderDangerButton()', () => {
      expect(global.renderToJSON(renderDangerButton())).toMatchSnapshot()
    })

    test('must match renderDangerSmallButton()', () => {
      expect(global.renderToJSON(renderDangerSmallButton())).toMatchSnapshot()
    })

    test('must match renderDangerLargeButton()', () => {
      expect(global.renderToJSON(renderDangerLargeButton())).toMatchSnapshot()
    })

    test('must match renderDangerDisabledButton()', () => {
      expect(global.renderToJSON(renderDangerDisabledButton())).toMatchSnapshot()
    })

    test('must match renderDangerIconButtonLeftAligned()', () => {
      expect(global.renderToJSON(renderDangerIconButtonLeftAligned())).toMatchSnapshot()
    })

    test('must match renderDangerIconButtonRightAligned()', () => {
      expect(global.renderToJSON(renderDangerIconButtonRightAligned())).toMatchSnapshot()
    })

    test('must match renderOutlineButton()', () => {
      expect(global.renderToJSON(renderOutlineButton())).toMatchSnapshot()
    })

    test('must match renderOutlineSmallButton()', () => {
      expect(global.renderToJSON(renderOutlineSmallButton())).toMatchSnapshot()
    })

    test('must match renderOutlineLargeButton()', () => {
      expect(global.renderToJSON(renderOutlineLargeButton())).toMatchSnapshot()
    })

    test('must match renderOutlineDisabledButton()', () => {
      expect(global.renderToJSON(renderOutlineDisabledButton())).toMatchSnapshot()
    })

    test('must match renderOutlineIconButtonLeftAligned()', () => {
      expect(global.renderToJSON(renderOutlineIconButtonLeftAligned())).toMatchSnapshot()
    })

    test('must match renderOutlineIconButtonRightAligned()', () => {
      expect(global.renderToJSON(renderOutlineIconButtonRightAligned())).toMatchSnapshot()
    })

    test('must match renderBlueButton()', () => {
      expect(global.renderToJSON(renderBlueButton())).toMatchSnapshot()
    })

    test('must match renderBlueSmallButton()', () => {
      expect(global.renderToJSON(renderBlueSmallButton())).toMatchSnapshot()
    })

    test('must match renderBlueLargeButton()', () => {
      expect(global.renderToJSON(renderBlueLargeButton())).toMatchSnapshot()
    })

    test('must match renderBlueDisabledButton()', () => {
      expect(global.renderToJSON(renderBlueDisabledButton())).toMatchSnapshot()
    })

    test('must match renderBlueIconButtonLeftAligned()', () => {
      expect(global.renderToJSON(renderBlueIconButtonLeftAligned())).toMatchSnapshot()
    })

    test('must match renderBlueIconButtonRightAligned()', () => {
      expect(global.renderToJSON(renderBlueIconButtonRightAligned())).toMatchSnapshot()
    })

    test('must match renderOrangeButton()', () => {
      expect(global.renderToJSON(renderOrangeButton())).toMatchSnapshot()
    })

    test('must match renderOrangeSmallButton()', () => {
      expect(global.renderToJSON(renderOrangeSmallButton())).toMatchSnapshot()
    })

    test('must match renderOrangeLargeButton()', () => {
      expect(global.renderToJSON(renderOrangeLargeButton())).toMatchSnapshot()
    })

    test('must match renderOrangeDisabledButton()', () => {
      expect(global.renderToJSON(renderOrangeDisabledButton())).toMatchSnapshot()
    })

    test('must match renderOrangeIconButtonLeftAligned()', () => {
      expect(global.renderToJSON(renderOrangeIconButtonLeftAligned())).toMatchSnapshot()
    })

    test('must match renderOrangeIconButtonRightAligned()', () => {
      expect(global.renderToJSON(renderOrangeIconButtonRightAligned())).toMatchSnapshot()
    })
  })
})
