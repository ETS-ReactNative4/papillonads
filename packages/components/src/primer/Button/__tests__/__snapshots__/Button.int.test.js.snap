// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Render must match renderBlueButton() 1`] = `
<button
  className="btn btn-blue"
  onClick={[Function]}
  type="button"
>
  Blue button
</button>
`;

exports[`<Button /> Render must match renderBlueDisabledButton() 1`] = `
<button
  aria-disabled="true"
  className="btn btn-blue"
  onClick={[Function]}
  type="button"
>
  Blue disabled button
</button>
`;

exports[`<Button /> Render must match renderBlueIconButtonLeftAligned() 1`] = `
<button
  className="btn btn-blue"
  onClick={[Function]}
  type="button"
>
  <mock-file
    aria-hidden="true"
    aria-label={null}
    className="mr-2"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
  Blue icon button
</button>
`;

exports[`<Button /> Render must match renderBlueIconButtonRightAligned() 1`] = `
<button
  className="btn btn-blue"
  onClick={[Function]}
  type="button"
>
  Blue icon button
  <mock-file
    aria-hidden="true"
    aria-label={null}
    className="ml-2"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
</button>
`;

exports[`<Button /> Render must match renderBlueLargeButton() 1`] = `
<button
  className="btn btn-large btn-blue"
  onClick={[Function]}
  type="button"
>
  Blue large button
</button>
`;

exports[`<Button /> Render must match renderBlueSmallButton() 1`] = `
<button
  className="btn btn-sm btn-blue"
  onClick={[Function]}
  type="button"
>
  Blue small button
</button>
`;

exports[`<Button /> Render must match renderDangerButton() 1`] = `
<button
  className="btn btn-danger"
  onClick={[Function]}
  type="button"
>
  Danger button
</button>
`;

exports[`<Button /> Render must match renderDangerDisabledButton() 1`] = `
<button
  aria-disabled="true"
  className="btn btn-danger"
  onClick={[Function]}
  type="button"
>
  Danger disabled button
</button>
`;

exports[`<Button /> Render must match renderDangerIconButtonLeftAligned() 1`] = `
<button
  className="btn btn-danger"
  onClick={[Function]}
  type="button"
>
  <mock-file
    aria-hidden="true"
    aria-label={null}
    className="mr-2"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
  Danger icon button
</button>
`;

exports[`<Button /> Render must match renderDangerIconButtonRightAligned() 1`] = `
<button
  className="btn btn-danger"
  onClick={[Function]}
  type="button"
>
  Danger icon button
  <mock-file
    aria-hidden="true"
    aria-label={null}
    className="ml-2"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
</button>
`;

exports[`<Button /> Render must match renderDangerLargeButton() 1`] = `
<button
  className="btn btn-large btn-danger"
  onClick={[Function]}
  type="button"
>
  Danger large button
</button>
`;

exports[`<Button /> Render must match renderDangerSmallButton() 1`] = `
<button
  className="btn btn-sm btn-danger"
  onClick={[Function]}
  type="button"
>
  Danger small button
</button>
`;

exports[`<Button /> Render must match renderDefaultButton() 1`] = `
<button
  className="btn"
  onClick={[Function]}
  type="button"
>
  Default button
</button>
`;

exports[`<Button /> Render must match renderDefaultDisabledButton() 1`] = `
<button
  aria-disabled="true"
  className="btn"
  onClick={[Function]}
  type="button"
>
  Default disabled button
</button>
`;

exports[`<Button /> Render must match renderDefaultIconButtonLeftAligned() 1`] = `
<button
  className="btn"
  onClick={[Function]}
  type="button"
>
  <mock-file
    aria-hidden="true"
    aria-label={null}
    className="mr-2"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
  Default icon button
</button>
`;

exports[`<Button /> Render must match renderDefaultIconButtonRightAligned() 1`] = `
<button
  className="btn"
  onClick={[Function]}
  type="button"
>
  Default icon button
  <mock-file
    aria-hidden="true"
    aria-label={null}
    className="ml-2"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
</button>
`;

exports[`<Button /> Render must match renderDefaultLargeButton() 1`] = `
<button
  className="btn btn-large"
  onClick={[Function]}
  type="button"
>
  Default large button
</button>
`;

exports[`<Button /> Render must match renderDefaultLinkButton() 1`] = `
<a
  className="btn"
  href="#url"
  onClick={[Function]}
  role="button"
>
  Default link button
</a>
`;

exports[`<Button /> Render must match renderDefaultSmallButton() 1`] = `
<button
  className="btn btn-sm"
  onClick={[Function]}
  type="button"
>
  Default small button
</button>
`;

exports[`<Button /> Render must match renderOrangeButton() 1`] = `
<button
  className="btn btn-orange"
  onClick={[Function]}
  type="button"
>
  Orange button
</button>
`;

exports[`<Button /> Render must match renderOrangeDisabledButton() 1`] = `
<button
  aria-disabled="true"
  className="btn btn-orange"
  onClick={[Function]}
  type="button"
>
  Orange disabled button
</button>
`;

exports[`<Button /> Render must match renderOrangeIconButtonLeftAligned() 1`] = `
<button
  className="btn btn-orange"
  onClick={[Function]}
  type="button"
>
  <mock-file
    aria-hidden="true"
    aria-label={null}
    className="mr-2"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
  Orange icon button
</button>
`;

exports[`<Button /> Render must match renderOrangeIconButtonRightAligned() 1`] = `
<button
  className="btn btn-orange"
  onClick={[Function]}
  type="button"
>
  Orange icon button
  <mock-file
    aria-hidden="true"
    aria-label={null}
    className="ml-2"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
</button>
`;

exports[`<Button /> Render must match renderOrangeLargeButton() 1`] = `
<button
  className="btn btn-large btn-orange"
  onClick={[Function]}
  type="button"
>
  Orange large button
</button>
`;

exports[`<Button /> Render must match renderOrangeSmallButton() 1`] = `
<button
  className="btn btn-sm btn-orange"
  onClick={[Function]}
  type="button"
>
  Orange small button
</button>
`;

exports[`<Button /> Render must match renderOutlineButton() 1`] = `
<button
  className="btn btn-outline"
  onClick={[Function]}
  type="button"
>
  Outline button
</button>
`;

exports[`<Button /> Render must match renderOutlineDisabledButton() 1`] = `
<button
  aria-disabled="true"
  className="btn btn-outline"
  onClick={[Function]}
  type="button"
>
  Outline disabled button
</button>
`;

exports[`<Button /> Render must match renderOutlineIconButtonLeftAligned() 1`] = `
<button
  className="btn btn-outline"
  onClick={[Function]}
  type="button"
>
  <mock-file
    aria-hidden="true"
    aria-label={null}
    className="mr-2"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
  Outline icon button
</button>
`;

exports[`<Button /> Render must match renderOutlineIconButtonRightAligned() 1`] = `
<button
  className="btn btn-outline"
  onClick={[Function]}
  type="button"
>
  Outline icon button
  <mock-file
    aria-hidden="true"
    aria-label={null}
    className="ml-2"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
</button>
`;

exports[`<Button /> Render must match renderOutlineLargeButton() 1`] = `
<button
  className="btn btn-large btn-outline"
  onClick={[Function]}
  type="button"
>
  Outline large button
</button>
`;

exports[`<Button /> Render must match renderOutlineSmallButton() 1`] = `
<button
  className="btn btn-sm btn-outline"
  onClick={[Function]}
  type="button"
>
  Outline small button
</button>
`;

exports[`<Button /> Render must match renderPrimaryButton() 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}
  type="button"
>
  Primary button
</button>
`;

exports[`<Button /> Render must match renderPrimaryDisabledButton() 1`] = `
<button
  aria-disabled="true"
  className="btn btn-primary"
  onClick={[Function]}
  type="button"
>
  Primary disabled button
</button>
`;

exports[`<Button /> Render must match renderPrimaryIconButtonLeftAligned() 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}
  type="button"
>
  <mock-file
    aria-hidden="true"
    aria-label={null}
    className="mr-2"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
  Primary icon button
</button>
`;

exports[`<Button /> Render must match renderPrimaryIconButtonRightAligned() 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}
  type="button"
>
  Primary icon button
  <mock-file
    aria-hidden="true"
    aria-label={null}
    className="ml-2"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
</button>
`;

exports[`<Button /> Render must match renderPrimaryLargeButton() 1`] = `
<button
  className="btn btn-large btn-primary"
  onClick={[Function]}
  type="button"
>
  Primary large button
</button>
`;

exports[`<Button /> Render must match renderPrimarySmallButton() 1`] = `
<button
  className="btn btn-sm btn-primary"
  onClick={[Function]}
  type="button"
>
  Primary small button
</button>
`;
